csm_web/frontend/src/components/data_export/ExportPage.tsx

```diff
@@ -0,0 +1,253 @@
+  };
+
+  return (
+    <div className="export-page-container">
+      <div className="export-page-config">
+        <div className="export-page-sidebar sidebar-left">{courseSelection}</div>
+        <div className="export-page-sidebar sidebar-right">{columnSelection}</div>
+      </div>
+      <ExportPagePreview courses={includedCourses} fields={includedFields} exportType={dataExportType} />
+      <div className="export-page-footer">
+        <button className="primary-btn" onClick={downloadData}>
+          Download Data
+        </button>
+      </div>
+    </div>
+  );
+};
+
+const PREVIEW_OPTIONS = [5, 10, 25, 50];
+
+interface ExportPagePreviewProps {
+  courses: number[];
+  fields: string[];
+  exportType: ExportType;
+}
+
+/**
+ * Preview of the exported data
+ */
+const ExportPagePreview = ({ exportType, courses, fields }: ExportPagePreviewProps) => {
+  const dataExportPreviewMutation = useDataExportPreviewMutation();
+  const [preview, setPreview] = useState<number>(10);
+  const [data, setData] = useState<string[][]>([]);
+
+  // automatically refresh on change
+  useEffect(() => {
+    refreshPreview();
+  }, [courses, fields, preview]);
+
+  const refreshPreview = () => {
+    dataExportPreviewMutation.mutate(
+      {
+        courses: courses,
+        fields: fields,
+        type: exportType,
+        preview: preview
+      },
+      {
+        onSuccess: dataPreview => {
+          setData(dataPreview);
+        }
+      }
+    );
+  };
+
+  const handlePreviewSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedValue = parseInt(e.target.value);
+    setPreview(selectedValue);
+  };
+
+  let dataTable = null;
+
+  if (data?.length > 1) {
+    // if has header and at least one row of content, display it
+    dataTable = (
+      <table className="export-preview-table">
+        <thead>
+          <tr className="export-preview-table-header">
+            {data?.length > 0 &&
+              data[0].map((cell, cellIdx) => (
+                <th key={cellIdx} className="export-preview-table-header-item">
+                  {cell}
+                </th>
+              ))}
+          </tr>
+        </thead>
+        <tbody>
+          {data.map(
+            (row, rowIdx) =>
+              rowIdx > 0 && (
+                <tr key={rowIdx}>
+                  {row.map((cell, cellIdx) => (
+                    <td key={cellIdx} className="export-preview-table-item">
+                      {cell}
+                    </td>
+                  ))}
+                </tr>
+              )
+          )}
+          {data?.length >= preview && (
+            <tr className="export-preview-table-more-row">
+              <td className="export-preview-table-more-row-item" colSpan={data[0].length}>
+                <i>(More rows clipped)</i>
+              </td>
+            </tr>
+          )}
+        </tbody>
+      </table>
+    );
+  } else {
+    // not enough data
+    dataTable = <i>Query returned no data.</i>;
```

Comment:
Modified to resolve that issue; it's a little bit hard to check for no returned data when >= 1 course is selected, since it's probably not the best to use preview data as a proxy for the real data (since preview data is supposed to be kinda dummy data).

So at the moment, it will download an empty CSV (with headers) if there is no data, and the button is grayed out if no courses are selected, to prevent the errors.